dm/config/task.go

```diff
@@ -775,7 +777,7 @@ func (c *TaskConfig) adjust() error {
 		instanceIDs[inst.SourceID] = i
 
 		switch c.TaskMode {
-		case ModeFull, ModeAll:
+		case ModeFull, ModeAll, ModeDump:
 			if inst.Meta != nil {
 				log.L().Warn("metadata will not be used. for Full mode, incremental sync will never occur; for All mode, the meta dumped by MyDumper will be used", zap.Int("mysql instance", i), zap.String("task mode", c.TaskMode))
 			}
```

Comment:
should we add ModeLoadSync? seems ModeLoadSync will also use the metadata file in source directory.

Reply:
ModeLoadSync can use metadata from syncers and it can use metadata file. So `inst.Meta` is empty or not is both okay?

Reply:
it's about how the upper layer uses engine, and I'm not sure about the behaviour of upper layer. lgtm now